pkg/config/validation/validation.go

```diff
@@ -2003,6 +2003,10 @@ var ValidateAuthorizationPolicy = registerValidateFunc("ValidateAuthorizationPol
 					}
 				}
 				for _, when := range rule.GetWhen() {
+					if when == nil {
```

Comment:
I am not sure how user can set nil with yaml

Reply:
```yaml
apiVersion: security.istio.io/v1beta1
kind: AuthorizationPolicy
metadata:
  name: httpbin
spec:
  action: ALLOW
  rules:
  - 
    when:
    - ~
```

But we also validate non-yaml (XDS, etc)